subprojects/core/src/integTest/groovy/org/gradle/api/tasks/ReproducibleArchivesIntegrationTest.groovy

```diff
@@ -106,7 +106,7 @@ class ReproducibleArchivesIntegrationTest extends AbstractIntegrationSpec {
             task tar(type: Tar) {
                 reproducibleFileOrder = true
                 preserveFileTimestamps = false
-                compression = '${compression}'
```

Comment:
🤔 I have zero idea how this was working with a `String` before. Checked the current docs, and no surprises there - we expect a `Compression` type, and everything checks out.

Reply:
Yeah, for Groovy we have some magic to translate strings to enums :D

Reply:
That's good to hear that this is a special case.
Can you give a go/nogo on this solution?

Reply:
Currently we only support direct usage of enums, so this is the only way. If we will add support for string version it will be with the deprecation.

So this solution is what we want. I would just keep `=` instead of using `.set()`